osu.Game/Graphics/Containers/UserDimContainer.cs

```diff
@@ -72,15 +70,13 @@ public UserDimContainer(bool isStoryboard = false)
             AddInternal(DimContainer = new Container { RelativeSizeAxes = Axes.Both });
         }
 
-        /// <summary>
-        /// Set the blur of the background in this UserDimContainer to our blur target instantly.
-        /// </summary>
-        /// <remarks>
-        /// We need to support instant blurring here in the case of changing beatmap backgrounds, where blurring shouldn't be from 0 every time the beatmap is changed.
-        /// </remarks>
-        public void ApplyInstantBlur()
+        public override void Add(Drawable drawable)
         {
-            background?.BlurTo(blurTarget, 0, Easing.OutQuint);
+            // We need to blur instantly here in the case of changing beatmap backgrounds, where blurring shouldn't be from 0 every time the beatmap is changed.
+            if (drawable is Background b)
+                b.BlurTo(blurTarget, 0, Easing.OutQuint);
```

Comment:
store to a field instead of using linq to find the background later.

Reply:
comment shouldn't mention outer usages (UserDimContainer should not be aware of beatmap usage)